ee/fleetctl/updates.go

```diff
@@ -320,6 +309,96 @@ func updatesTimestampFunc(c *cli.Context) error {
 	return nil
 }
 
+func updatesRotateCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "rotate",
+		Usage:     "Rotate signing keys",
+		ArgsUsage: "<role>",
+		Description: `Rotate the signing keys used for updates metadata signing. This should be used when keys are compromised or expiring.
+		
+role must be one of ['root', 'targets', 'timestamp', 'snapshot']
+		`,
+		Flags:  updatesFlags(),
+		Action: updatesRotateFunc,
+	}
+}
+
+func updatesRotateFunc(c *cli.Context) error {
+	if c.NArg() != 1 {
+		return errors.New("role must be provided")
+	}
+	role := c.Args().Get(0)
+
+	repo, err := openRepo(c.String("path"))
+	if err != nil {
+		return err
+	}
+	store, err := openLocalStore(c.String("path"))
+	if err != nil {
+		return err
+	}
+
+	if err := checkKeys(c.String("path"),
+		"root",
+		"targets",
+		"snapshot",
+		"timestamp",
+	); err != nil {
+		return err
+	}
+
+	// Get old keys for role
+	keys, err := store.GetSigningKeys(role)
+	if err != nil {
+		return errors.Wrap(err, "get keys for role")
+	}
+
+	// Generate new key for role
+	if err := updatesGenKey(repo, role); err != nil {
+		return err
+	}
+
+	// Delete old keys for role
+	for _, key := range keys {
+		id := key.IDs()[0]
+		err := repo.RevokeKeyWithExpires(role, id, time.Now().Add(rootExpirationDuration))
+		if err != nil {
+			// go-tuf keeps keys around even after they are revoked from the manifest. We can skip
+			// tuf.ErrKeyNotFound as these represent keys that are not present in the manifest and
+			// so do not need to be revoked.
+			if !errors.As(err, &tuf.ErrKeyNotFound{}) {
+				return errors.Wrap(err, "revoke key")
+			}
+		}
+	}
```

Comment:
Should you delete old keys at the end in case anything else fails?

Reply:
The keys need to be revoked before generating the new metadata. I'll add a "rollback" routine to try to handle the case of anything failing after this step.